services/outputhost/cgcache.go

```diff
@@ -444,7 +444,23 @@ func (cgCache *consumerGroupCache) manageConsumerGroupCache() {
 	}
 }
 
-// refreshCgCache contacts metadata to get all the open extents and then loads them if needed
+// refreshCgCacheNoLock is a routine which is called without the extMutex held.
+// It takes the mutex and in turn refreshes the cg by contactingmetadata to
```

Comment:
this routine, refreshCgCacheNoLock() gets called without the lock held..i.e, the caller doesn't hold the lock and it needs to be taken inside.. whereas the refreshCgCache() is the one where the caller already has the lock and a lock is not needed inside the routine..